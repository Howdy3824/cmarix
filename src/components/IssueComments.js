import React from 'react';
import IssueComment from './IssueComment';

export default function IssueComments({ comments = [] }) {
  return (
    <ul className="issue-detail__comments">
      {comments.map(comment =>
        <li key={comment.id}>
          <IssueComment comment={comment}/>
        </li>
      )}
    </ul>
  );
}


