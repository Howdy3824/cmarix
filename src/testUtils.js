import toJson from 'enzyme-to-json';

export function afterPromises(done, fn) {
  setTimeout(() => {
    try {
      fn();
      done();
    } catch (e) {
      fail();
    }
  }, 1);
}

export function checkSnapshot(tree) {
  expect(toJson(tree)).toMatchSnapshot();
}