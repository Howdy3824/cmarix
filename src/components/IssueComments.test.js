import React from 'react';
import { shallow } from 'enzyme';
import { checkSnapshot } from '../utils/testUtils';
import IssueComments from './IssueComments';

it('renders with no comments', () => {
  const tree = shallow(<IssueComments/>);
  checkSnapshot(tree);
});

it('renders comments', () => {
  const comment = {
    user: {
      login: "somebody",
      avatar_url: "http://foo"
    },
    body: "a comment"
  };
  const comments = [{...comment, id: 1}, {...comment, id: 2}];
  const tree = shallow(
    <IssueComments comments={comments}/>
    );
  checkSnapshot(tree);
});