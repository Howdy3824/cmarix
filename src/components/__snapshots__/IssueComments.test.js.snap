exports[`test renders comments 1`] = `
<ul
  className="issue-detail__comments">
  <li>
    <IssueComment
      comment={
        Object {
          "body": "a comment",
          "id": 1,
          "user": Object {
            "avatar_url": "http://foo",
            "login": "somebody",
          },
        }
      } />
  </li>
  <li>
    <IssueComment
      comment={
        Object {
          "body": "a comment",
          "id": 2,
          "user": Object {
            "avatar_url": "http://foo",
            "login": "somebody",
          },
        }
      } />
  </li>
</ul>
`;

exports[`test renders with no comments 1`] = `
<ul
  className="issue-detail__comments" />
`;
